examples/crm/src/sales/TransferAdminRoleModal.tsx

```diff
@@ -0,0 +1,114 @@
+import Alert from '@mui/material/Alert';
+import Autocomplete from '@mui/material/Autocomplete';
+import Button from '@mui/material/Button';
+import CircularProgress from '@mui/material/CircularProgress';
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
+import DialogTitle from '@mui/material/DialogTitle';
+import Skeleton from '@mui/material/Skeleton';
+import { default as MuiTextField } from '@mui/material/TextField';
+import { useState } from 'react';
+import {
+    useDataProvider,
+    useGetIdentity,
+    useGetList,
+    useRefresh,
+} from 'react-admin';
+import { useNavigate } from 'react-router';
+import { USER_STORAGE_KEY } from '../authProvider';
+import { CustomDataProvider } from '../dataProvider';
+import { Sale } from '../types';
+type TransferAdminRoleModalProps = {
+    open: boolean;
+    onClose(): void;
+};
+
+export function TransferAdminRoleModal({
+    open,
+    onClose,
+}: TransferAdminRoleModalProps) {
+    const { identity } = useGetIdentity();
+    const dataProvider = useDataProvider<CustomDataProvider>();
+    const refresh = useRefresh();
+    const navigate = useNavigate();
+    const {
+        data: sales,
+        error,
+        isPending,
+    } = useGetList('sales', {
+        filter: { administrator: false },
+        pagination: {
+            page: 1,
+            perPage: 1000,
+        },
+        sort: { field: 'first_name', order: 'ASC' },
+    });
+    const [to, setTo] = useState<Sale | null>(null);
+    const [loading, setLoading] = useState(false);
+
+    const handleTransfer = () => {
+        if (!identity || !to) {
+            return;
+        }
+
+        setLoading(true);
+        void dataProvider
+            .transferAdministrator(identity.id, to.id)
+            .then(updatedUser => {
+                localStorage.setItem(
+                    USER_STORAGE_KEY,
+                    JSON.stringify(updatedUser?.data)
+                );
+            })
+            .finally(() => {
+                onClose();
+                setLoading(false);
+                navigate('/');
```

Comment:
Why change page?

Reply:
This is requested in the card, as the current user is no longer an administrator once they transfered the role